### PR TITLE
Deposit contract page [#1553]

### PR DESCRIPTION
**Pull request by ryancreatescopy**
## Description

Create a page that allows users to double check the deposit contract address for eth2 staking. 

THIS IS WIP AND A DUMMY ADDRESS – DON'T SEND ANYTHING TO THIS ADDRESS.

## TODO

- [ ] @ryancreatescopy fill in correct address & URLs (once available)
- [x] @ryancreatescopy confirm if we want a banner notification on /eth2/ pages
- [x] @ryancreatescopy confirm plans around an /eth2/staking/ page (& write if needed)
- [x] @samajammin add `<BannerNotification />` on all /eth2/ pages

## Related Issue

#1553

### CODE REVIEW

File: src/content/eth2/index.md [hunk 1]
```diff
@@ -74,11 +74,13 @@ Eth2 is designed to scale without sacrificing performance, security or decentral
...
+
 Staking is the process of committing your ETH to become a validator for Ethereum. A validator runs software that confirms transactions and, when chosen, creates new blocks in the chain. To become a full validator you'll need 32 ETH. However there will be opportunities to offer a smaller stake by joining staking pools.
 
 There will be thousands of validators. And they will keep the network secure by processing transactions and making sure everyone agrees on the same data.
 
-Ethereum today is a [proof-of-work blockchain](/en/learn/#proof-of-work-and-mining). It relies on miners to keep the network secure and in sync by devoting a great amount of computing power to creating new blocks. Proof-of-stake keeps the network secure but replaces energy consumption with a financial commitment.
+Looking to learn more about staking? Find the [staking smart contract address](/eth2/staking-address/).
```
Comment (samajammin) [inline src/content/eth2/index.md:83]: @ryancreatescopy - FYI, I've added this link & copy to the main Eth2 page. I welcome your edits 😄

File: src/pages/eth2/staking-address.js [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+
+const SyledCheckbox = styled(Checkbox)`
+  display: flex;
+  min-height: 3rem;
+  @media (max-width: ${(props) => props.theme.breakpoints.m}) {
+    min-height: 4rem;
+  }
+`
+
+// TODO confirm/update
```
Comment (samajammin) [inline src/pages/eth2/staking-address.js:148]: @ryancreatescopy FYI moved the address here & a function to "chunk" it so we only ever define the address in one place.

File: src/pages/eth2/staking-address.js [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+          <br /> Use this page to confirm you’re using the correct deposit
+          address.
+        </Subtitle>
+        <StyledCard
+          emoji=":octagonal_sign:"
+          title="This is not where you stake"
+          description="To stake your ETH in Eth2 you must use the launchpad and follow the instructions. Sending ETH to this address will not make you a staker and will result in a failed transaction."
+        >
+          <ButtonRow>
+            {/* TODO add URL */}
```
Comment (samajammin) [inline src/pages/eth2/staking-address.js:199]: TODO

File: src/pages/eth2/staking-address.js [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+            <StyledButton to="#">Stake using launchpad</StyledButton>
+            {/* TODO add URL */}
```
Comment (samajammin) [inline src/pages/eth2/staking-address.js:201]: TODO

File: src/pages/eth2/staking-address.js [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+                          address
+                        </div>
+                      )}
+                    </CopyButton>
+                  )}
+                </CopyToClipboard>
+              </>
+            )}
+            <Warning emoji=":warning:">
+              {/* TODO add URL */}
```
Comment (samajammin) [inline src/pages/eth2/staking-address.js:309]: TODO

**Review by samajammin — CHANGES_REQUESTED** [review]
A few TODOs (as expected) to update the address & URLs.

**Comment by samajammin** [comment]
@ryancreatescopy looks great! A couple things to note re: my updates:
- I moved the page, it's now located within the Eth2 directory (i.e. /eth2/staking-address/)
- I updated the `Button` component to `ButtonLink` as we discussed... you can safely ignore most of these file edits

One question - do we intend to create an /eth2/staking/ page in tandem with this release? I think we should provide a more in-depth resource on the how & why of staking. Our current content on the main Eth2 page (under "Proof of stake" is pretty sparse). If we do this, I'd probably move the address page (i.e. /eth2/staking/address/). LMK what our plans are around that.

**Comment by ryancreatescopy** [comment]
> @ryancreatescopy looks great! A couple things to note re: my updates:
> 
> * I moved the page, it's now located within the Eth2 directory (i.e. /eth2/staking-address/)
> * I updated the `Button` component to `ButtonLink` as we discussed... you can safely ignore most of these file edits
> 
> One question - do we intend to create an /eth2/staking/ page in tandem with this release? I think we should provide a more in-depth resource on the how & why of staking. Our current content on the main Eth2 page (under "Proof of stake" is pretty sparse). If we do this, I'd probably move the address page (i.e. /eth2/staking/address/). LMK what our plans are around that.

I do intend to create a staking page – I'll start drafting this considering we wont be shipping this page for a little while. See a proposed structure here: https://www.notion.so/efdn/Scope-out-Eth2-v2-section-f8d938bae9064773a9f13c28f1c1ca41#572f96eb0cab41ffb8bd092681801790
I can also confirm that we do want a banner notification on the other ETH2 pages

**Comment by ryancreatescopy** [comment]
@samajammin can we make the left column scrollable but the right column sticky so the check address component remains in view?

**Comment by ryancreatescopy** [comment]
image

@samajammin The margin-bottom on the banner on mobile is a little tight to the top of the page. I reckon we need 2rem of margin.

**Review by samajammin — APPROVED** [review]
Merging this for the sake of the far-reaching style & component changes. I'll create a follow-up PR for deploying the page & updated address.

**Comment by samajammin** [comment]
@ryancreatescopy @jjmstark - note the src/data/addresses.js file that now hosts the (dummy) contact address. That will be the file to keep a close eye on in future deploys.

**Pull request merged**